NOTES/HowStuffWorks.md

```diff
@@ -0,0 +1,118 @@
+
+### page reloading
+Ensure that you are running the app webserver in development mode.
+
+Then run another server using `npm run dev-server`.
+It wil rebuild while watching for file changes, and it wil trigger your page to reload afterwards.
+
+### hot reloading
+Ensure that you are running the app webserver in development mode.
+
+Then run another server using `npm run hot-dev-server`.
+It wil rebuild while watching for file changes, and it wil update the currently shown and affected component(s) while keeping their state.
+Note this is experimental, and in some cases you'll need to refresh manually.
+
+
+## How the routes work.
+In React after opening the app and going to some page, there is no actual HTML loaded from the server. React app just replaces a component that acts as a page, including any child components.
+But you'd like to have the URL reflect this, and allow user to use browser history (back/forward). A router takes care of these things. (package react-router)
+
+In this case, the root of your app is not the Application React component.
+This starts at `/lib/server.js` which wil use `/config/app.jsx` which instantiates the router and ultimately uses `/app/mainRoutes.jsx` to load routes.
+You'll find that all pages are subroutes within the `app` route, which instantiates `/app/Application/index.jsx`, which contains a `RouteHander` component that inserts subroute output.
+
+
+## How the server JSON API works.
+The `/lib/server.js` serves the application, but it also serves the API URLs that the stores talk with.
+It initializes two databases (todolist and todoitem) once, and then continues to listen for GET/POST requests on specific URLs.
+
+
+## How the stores work.
+As in Flux; the Stores affect the React components. (package items-store) And the stores talk to the JSON API. (package superagent)
+
+### Stores setup
+The stores are constructed as such: startpoint is `/config/app.jsx` which wil use `/app/mainStores.js`. This then:
+
+- defines routines to handle JSON API read/writes (package superagent), and 
+- sets up a queue that only allows one REST request at a time, and aggregates subsequent changes. 
+  These are then sent as one request.. :question:
+- and ultimately constructs the two respective `ItemStore` objects (based on `/app/mainStoresDescriptions.js`) to which it assigns these routines, queue, and the initial data. This initial data may have been inserted when `/app/prerender.html`.
+- These are then exported back to `/config/app.jsx`, along with a store named Router. This store just holds a `transition` value which is later read in `/Application/index.jsx` to show "loading...' in UI.
+
+### Stores in use
+
+todo :question: this section needs review and should be extended.
+
+Note on `Application.update()`: normally components should not access the stores except for reading in `getState()`, as demonstrated with the Router store. Everything else should be done with actions.
```

Comment:
@sokra i feel that you'll probably have something to add on the `How the stores work` section above. You grasp that way better than me;

- How is it triggered to refresh everything from the server,
- how does it optimize this diff when loading from server - if at all,
- how does it propagate changes when the user edits items, and
- how do values travel before ending up in some components `render()` ?
- how does the queue combine multiple requests? I cant imagine subsequent add/remove/edits would result in just one rest call.. do they?

Amongst other paths that the store data travels that you may know of that i dont..